numba/targets/arraymath.py

```diff
@@ -3117,6 +3117,28 @@ def diff_impl(a, n=1):
     return diff_impl
 
 
+@overload(np.array_equal)
+def np_array_equal(a, b):
+
+    if not (type_can_asarray(a) and type_can_asarray(b)):
+        raise TypingError('Both arguments to "array_equals" must be array-like')
+
+    if isinstance(a, (types.Integer, types.Boolean, types.Number)) and \
+       isinstance(b, (types.Integer, types.Boolean, types.Number)):
```

Comment:
Recall that `types.Integer` inherits from `types.Number`.
How about:
```python
accepted = (types.Boolean, types.Number)
if isinstance(a, accepted) and isinstance(b, accepted):
```

Reply:
Oh, didn't know about that. I've pushed the new version.